roxctl/common/tls.go

```diff
@@ -88,69 +46,57 @@ func getServerName(endpoint string) (string, error) {
 	return serverName, nil
 }
 
-func tlsConfigOptsForCentral(logger logger.Logger) (*clientconn.TLSConfigOptions, error) {
+func tlsConfigOptsForCentral() (*clientconn.TLSConfigOptions, error) {
 	_, serverName, _, err := ConnectNames()
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing central endpoint")
 	}
 
-	var dialContext func(ctx context.Context, addr string) (net.Conn, error)
-
-	skipVerify := flags.SkipTLSValidation() != nil && *flags.SkipTLSValidation()
-	var roots *x509.CertPool
-	var customVerifier tlscheck.TLSCertVerifier
-	var ca []byte
-	if skipVerify {
-		if flags.CAFile() != "" {
-			logger.WarnfLn("--insecure-skip-tls-verify has no effect when --ca is set")
-		}
-	} else {
-		customVerifier = &insecureVerifierWithError{
-			logger: logger,
-		}
-		if flags.CAFile() != "" {
-			var err error
-			if ca, err = os.ReadFile(flags.CAFile()); err != nil {
-				return nil, errors.Wrap(err, "failed to parse CA certificates from file")
-			}
-		} else {
-			// Read the CA from the central secret.
-			if flags.UseKubeContext() {
-				_, core, namespace, err := getConfigs()
-				if err != nil {
-					return nil, err
-				}
-				var warn error
-				// Proceed with no CA on error. Return the error as warning later.
-				ca, warn = getCentralCA(context.Background(), core, namespace)
-				if warn != nil {
-					logger.WarnfLn("Failed to read the central CA: %v", warn)
-				}
-			}
-		}
+	opts := &clientconn.TLSConfigOptions{
+		ServerName:         serverName,
+		InsecureSkipVerify: flags.SkipTLSValidation() != nil && *flags.SkipTLSValidation(),
 	}
 
-	if ca != nil {
-		roots = x509.NewCertPool()
-		if !roots.AppendCertsFromPEM(ca) {
-			return nil, errors.Errorf("CA certificates file %s contains no certificates!", flags.CAFile())
+	if !opts.InsecureSkipVerify {
+		if opts.RootCAs, err = getCertPool(); err != nil {
+			return nil, err
 		}
 	}
+
 	if flags.UseKubeContext() {
-		dialContext = getForwardingDialContext()
+		opts.DialContext = getForwardingDialContext()
 	}
 
-	return &clientconn.TLSConfigOptions{
-		ServerName:         serverName,
-		InsecureSkipVerify: skipVerify,
-		CustomCertVerifier: customVerifier,
-		RootCAs:            roots,
-		DialContext:        dialContext,
-	}, nil
+	return opts, nil
+}
+
+func getCertPool() (*x509.CertPool, error) {
+	roots := x509.NewCertPool()
```

Comment:
`NewCertPool` returns a new, empty `CertPool`. I think we should use `SystemCertPool` to honor installed certs. or we can use `NewCertPool` as fallback when `SystemCertPool` returns an error 
 
```suggestion
	roots, err := x509.SystemCertPool()
	if err != nil {
		return nil, errors.Wrap(err, "could not get system certs pool")
	}
```

```go
	roots, err := x509.SystemCertPool()
	if err != nil {
		roots = x509.NewCertPool()
	}
```

See: 
- https://github.com/stackrox/central-login/pull/436